content/docs/state-and-lifecycle.md

```diff
@@ -302,84 +302,84 @@ ReactDOM.render(
+4) Co sekundę przeglądarka wywołuje metodę `tick()`. W jej ciele komponent `Clock` żąda aktualizacji UI poprzez wywołanie metody `setState()`, przekazując jako argument obiekt z aktualnym czasem. Dzięki wywołaniu `setState()` React wie, że zmienił się stan i że może ponownie wywołać metodę `render()`, by dowiedzieć się, co powinno zostać wyświetlone na ekranie. Tym razem wartość zmiennej `this.state.date` w ciele metody `render()` będzie inna, odpowiadająca nowemu czasowi - co React odzwierciedli w drzewie DOM.
 
-5) If the `Clock` component is ever removed from the DOM, React calls the `componentWillUnmount()` lifecycle method so the timer is stopped.
+5) Jeśli kiedykolwiek komponent `Clock` zostanie usunięty z drzewa DOM, React wywoła na nim metodę cyklu życia o nazwie `componentWillUnmount()`, zatrzymując tym samym timer.
 
-## Using State Correctly {#using-state-correctly}
+## Poprawne używanie stanu {#using-state-correctly}
 
-There are three things you should know about `setState()`.
+Są trzy rzeczy, które musisz wiedzieć o `setState()`.
 
-### Do Not Modify State Directly {#do-not-modify-state-directly}
+### Nie modyfikuj stanu bezpośrednio {#do-not-modify-state-directly}
 
-For example, this will not re-render a component:
+Na przykład, poniższy kod nie spowoduje ponownego wyrenderowania komponentu:
 
 ```js
-// Wrong
-this.state.comment = 'Hello';
+// Źle
+this.state.comment = 'Witam';
 ```
 
-Instead, use `setState()`:
+Zamiast tego używaj `setState()`:
 
 ```js
-// Correct
-this.setState({comment: 'Hello'});
+// Dobrze
+this.setState({comment: 'Witam'});
 ```
 
-The only place where you can assign `this.state` is the constructor.
+Jedynym miejscem, w którym wolno Ci użyć `this.state` jest konstruktor klasy.
 
-### State Updates May Be Asynchronous {#state-updates-may-be-asynchronous}
+### Aktualizacje stanu mogą dziać się asynchroniczne {#state-updates-may-be-asynchronous}
 
-React may batch multiple `setState()` calls into a single update for performance.
+React może zgrupować kilka wywołań metody `setState()` w jedną paczkę w celu zwiększenia wydajności aplikacji.
 
-Because `this.props` and `this.state` may be updated asynchronously, you should not rely on their values for calculating the next state.
+Z racji tego, że zmienne `this.props` i `this.state` mogą być aktualizowane asynchronicznie, nie powinno się polegać na ich wartościach przy obliczaniu nowego stanu.
 
-For example, this code may fail to update the counter:
+Na przykład, poniższy kod może nadpisać licznik błędną wartością:
```

Comment:
Nie wiem czy przetłumaczenie nazwy klucza z obiektu nie będzie odrobnę mylące. W angielskim to słowo jest dokładnie takie samo, wiec dokumentacja nie prezentuje go z żadnym wyróżnieniem. Proponuję zostawić counter w "ciapkach" .

`counter`
```suggestion
Na przykład, poniższy kod może nadpisać `counter` błędną wartością:
```

Reply:
Zasugerowałem się wersją angielską, która faktycznie nie wyróżnia tego słowa w żaden sposób. I właśnie dlatego je przetłumaczyłem, żeby czytelni wiedział, że chodzi o "licznik", a nie o jakąś tam zmienną "counter". Licznik jako pewna "funkcjonalność" czy "cecha" komponentu.

Reply:
Może w takim razie chociaż `(ang. counter)` obok? Dla kogoś, kto nie zna angielskiego może tam brakować logicznego połączenia.

Reply:
W takim razie już lepiej dodać te ciapki, żeby nie było wątpliwości i zbędnego szumu ;-) Dodam.